### PR TITLE
[Tuning] rocm 4.4 update for develop

### PR DESCRIPTION
**Pull request by cderb**

### CODE REVIEW

**Comment by junliume** [comment]
@cderb CI has some interesting issues:
> [2021-08-31T19:41:23.263Z] FAILED: pdb.FindRecord(ctx): /var/jenkins/workspace/MLLibs_MIOpen_tuning-rocm-4.4/test/embed_sqlite.cpp: 81
[2021-08-31T19:41:23.263Z] 
[2021-08-31T19:41:23.263Z] warning: core file may not match specified executable file.

The following tests FAILED:
>[2021-08-31T19:27:42.458Z] 	 18 - test_embed_sqlite (Failed)
[2021-08-31T19:27:42.458Z] 	 58 - test_conv_embed_db (Failed)

**Comment by atamazov** [comment]
I'll run performance tests as soon as CI passed.

**Comment by junliume** [comment]
> I'll run performance tests as soon as CI passed.

CI issues are reproducible so we may need to fix them first @cderb @JehandadKhan

**Comment by atamazov** [comment]
`test_conv_embed_db` is consistently failing.

**Comment by atamazov** [comment]
I will merge recent develop and restart CI.

**Comment by atamazov** [comment]
CI run #5 (with `fix test_conv_embed_db`)  still shows test_conv_embed_db failure.

CI run #8 failed with `filesystem::recursive_directory_iterator` error during "Fp32 **OpenCL** gfx90a". Maybe this is related to missing gfx90a databases for OpenCL backend?

Restarted.

**Comment by atamazov** [comment]
It seems like the `filesystem::recursive_directory_iterator` problem first happened a while ago, see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/701#issuecomment-894504863.

**Review by junliume — APPROVED** [review]

**Comment by junliume** [comment]
Issue #1161 logged and will keep track

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -562,7 +562,8 @@ set(IMPLICITGEMM_ARGS ${MIOPEN_TEST_FLOAT_ARG})
 
 # ./bin/MIOpenDriver conv -n 128 -c 1024 -H 14 -W 14 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
 # MIOPEN_DEBUG_CONV_IMMED_FALLBACK=0
-if(MIOPEN_EMBED_DB)
+# WORKAROUND_ISSUE_1061 : not MIOPEN_TEST_VEGA
+if(MIOPEN_EMBED_DB AND NOT MIOPEN_TEST_VEGA)
```
Comment (atamazov) [inline test/CMakeLists.txt:566]: Please see https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1161#issuecomment-922387281

**Comment by atamazov** [comment]
I am running performance tests on Vega20 right now.

**Comment by atamazov** [comment]
[Intermediate testing results] Unfortunately I see performance & correctness problems with FP32 on gfx906_60. Analysis is in progress.

An example of correctness problem: duplicated algorithm in find-db (file: gfx906_60.HIP.fdb.txt)
```
128-17-17-7x1-128-17-17-128-3x0-1x1-1x1-0-NCHW-FP32-F=
    miopenConvolutionFwdAlgoImplicitGEMM:
        ConvHipImplicitGemmV4R1Fwd,0.979353,0,miopenConvolutionFwdAlgoImplicitGEMM,128x17x17x7x1x128x17x17x128xNCHWxFP32x3x0x1x1x1x1x1xF;
    miopenConvolutionFwdAlgoImplicitGEMM:
        ConvAsmImplicitGemmV4R1DynamicFwd,0.979353,0,miopenConvolutionFwdAlgoImplicitGEMM,128x17x17x7x1x128x17x17x128xNCHWxFP32x3x0x1x1x1x1x1xF;
    miopenConvolutionFwdAlgoWinograd:
        ConvBinWinogradRxSf3x2,1.11983,0,miopenConvolutionFwdAlgoWinograd,128x17x17x7x1x128x17x17x128xNCHWxFP32x3x0x1x1x1x1x1xF;
    miopenConvolutionFwdAlgoDirect:
        ConvDirectNaiveConvFwd,20.013,0,miopenConvolutionFwdAlgoDirect,128x17x17x7x1x128x17x17x128xNCHWxFP32x3x0x1x1x1x1x1xF;
    miopenConvolutionFwdAlgoGEMM:
        GemmFwdRest,9081.04,1035776,miopenConvolutionFwdAlgoGEMM,128x17x17x7x1x128x17x17x128xNCHWxFP32x3x0x1x1x1x1x1xF
```
Two records that refer to the same algorithm, `miopenConvolutionFwdAlgoImplicitGEMM:`. Notice that time is identical (`0.979353`); perhaps this is the implicit reason of the issue.

**Comment by atamazov** [comment]
Maybe it is enough to revert only find-db changes. I am investigating this possibility.

**Comment by atamazov** [comment]
> Maybe it is enough to revert only find-db changes. I am investigating this possibility.

No. There is some FP32 perf regression (~23%) due to missing perf-db records (many "Perf Db: record not found for..." messages).

revert-tuning-rocm-4.4-fordev   - ONLY FIND-DB REVERTED | - | - | - | -
-- | -- | -- | -- | --
ROCm 4.0, gfx906/60 | - | Performance gain, based on sum of all times/directions | - | -
. | - | GPU time | Wall time | Aux Wall time
FP32 | Default find mode | 0.77 | 0.98 | 0.92
FP32 OpenCL BE | Default find mode | 0.77 | 0.96 | 1.02

FP16/BF16 not tested. Let's revert all.

**Pull request merged**